chainer/functions/activation/maxout.py

```diff
@@ -1,108 +1,39 @@
-import numpy
+from chainer.functions.array import reshape
+from chainer.functions.math import minmax
 
-from chainer import cuda
-from chainer import function
-from chainer.utils import type_check
 
+def maxout(x, num_channel, axis=1):
+    """Maxout activation function
 
-def _as_mat(x):
-    if x.ndim == 2:
-        return x
-    return x.reshape(len(x), -1)
+    It accepts an input tensor ``x``, reshapes the ``axis`` dimension
+    (say the size being ``M * num_channel``) into two dimensional tensor
```

Comment:
I use `pool_size` because the author write as "pool across channels" in the original paper.